### PR TITLE
Fix removing a disk from a Vm

### PR DESCRIPTION
**Pull request by borod108**
Make remove disk work both through the API and the UI.

Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1692801

### CODE REVIEW

**Comment by borod108** [comment]
@agrare @Ladas do you know why the travis fails? its strange - fails only for ruby 2.4.5

File: app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4.rb [hunk 1]
```diff
@@ -822,21 +822,16 @@ def prepare_vm_disk_attachment(disk_spec, storage_spec)
...
-
       disk_specs.each do |disk_spec|
         disk_spec = disk_spec.with_indifferent_access
-        attachment_service = disk_attachment_by_disk_name_hash[disk_spec['disk_name']]
-        unless attachment_service
-          raise "no disk with the name #{disk_spec['disk_name']} is attached to the vm: #{vm_service.get.name}"
-        end
+        attachment_service = attachments_service.attachment_service(disk_spec['disk_name'])
         attachment_service.remove(:detach_only => !disk_spec['delete_backing'])
+      rescue OvirtSDK4::NotFoundError
```
Comment (Ladas) [inline app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4.rb]: @borod108 the ci fails becuse you've placed the rescue badly either put it to `begin rescue end` or if it supposed to be a whole method rescue, put it out of the block

so
```
def xy

array.each do |a|
end

rescue Class
end
```

the rescue can;t be nested to the `each` block

Comment (borod108) [reply app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4.rb]: Thank you!

Comment (borod108) [reply app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4.rb]: wow this is actually cool, it did work for another version of ruby, and actually did work in practice!

**Comment by borod108** [comment]
@masayag please merge

**Comment by simaishi** [comment]
Hammer backport details:

```
$ git log -1
commit 47b40fc87653a86d357bf620653a34e77f3e3224
Author: Moti Asayag <masayag@redhat.com>
Date:   Tue Apr 2 10:22:35 2019 +0300

    Merge pull request #353 from borod108/bugs/1692801disk_removal
    
    Fix removing a disk from a Vm
    
    (cherry picked from commit e68ae01d6413cbf229de11f8996f6b4d88c607af)
    
    Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1695629
```

**Pull request merged**